OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs

```diff
@@ -115,9 +115,39 @@ void CreateAndJoin()
 			var settings = new ServerSettings(Game.Settings.Server);
 
 			// Create and join the server
-			Game.CreateServer(settings);
+			try
+			{
+				Game.CreateServer(settings);
+			}
+			catch (System.Net.Sockets.SocketException)
+			{
+				// Console.WriteLine("Couldn't listen on port {0}, socket already in use?", settings.ListenPort);
+				Ui.OpenWindow("CREATESERVERFAIL_PANEL", new WidgetArgs()
```

Comment:
Thanks. I'm now using CancelPrompt(). It's a much smaller diff now. I'm still glad I did it the other way first because it forced me to figure out how the prompts work.